pilot/lib/pilot/build_manager.rb

```diff
@@ -366,6 +366,9 @@ def expire_previous_builds(build)
     # If there are multiple teams, infer the provider from the selected team name.
     # If there are fewer than two teams, don't infer the provider.
     def transporter_for_selected_team(options)
+      # Ensure that `Spaceship::ConnectAPI.token` is set correctly, if required
+      login
```

Comment:
Instead of here, I really want to **start upload** with `login` **always** in [this line](https://github.com/fastlane/fastlane/blob/master/pilot/lib/pilot/build_manager.rb#L17) Please let me know if you want to have login in the start upload or in this transporter method


```ruby
# from this...

# Only need to login before upload if no apple_id was given      
# 'login' will be deferred until before waiting for build processing      
should_login_in_start = options[:apple_id].nil?      
start(options, should_login: should_login_in_start)

# to this...
start(options, should_login: true)

# If we go with approach then the login code will be in one place and maintenance will be easy too... I also notice, Deliver always do login on start so it will bring consistency too 😇
```